src/Eccube/Repository/PageLayoutRepository.php

```diff
@@ -138,6 +138,10 @@ public function getByUrl(DeviceType $DeviceType, $url)
             ))
             ->getSingleResult();
 
+        if(count($ownResult->getBlockPositions()) && ($url == 'preview') && ($page == 'homepage')) {
```

Comment:
at homepage, preview mode dont get Blocks = 'all page', because blocks = 'all page' be saved in preview mode before.
with other pages, preview mode not save Blocks= 'all page'.